R/od-funs.R

```diff
@@ -271,47 +272,65 @@ line2pointsn <- function(l){
 #' rf_no_id$id # [1] "1" "2" "3" "4"
 #' rf_with_id = line2route(l, l_id = "All")
 #' rf_with_id$id # [1] 38 10 44
+#' # demo with parallel version spliting route API requests over multiple CPU processes
+#' rf_parr <- line2route(l = l, parallel = TRUE)
 #' }
-line2route <- function(l, route_fun = "route_cyclestreet", n_print = 10, list_output = FALSE, l_id = NA, ...){
-
+line2route <- function(l, route_fun = "route_cyclestreet", n_print = 10, list_output = FALSE, l_id = NA, parallel = FALSE, ...){
   FUN <- match.fun(route_fun)
   ldf <- line2df(l)
+  n_ldf <- nrow(ldf)
 
-  if(list_output){
-    r <- as.list(rep(NA, length(l)))
+  error_fun <- function(e){ warning(paste("Fail for line number", i)) }
 
-    for(i in 1:nrow(ldf)){
-      tryCatch({
-        r[[i]] <- FUN(from = c(ldf$fx[i], ldf$fy[i]), to = c(ldf$tx[i], ldf$ty[i]), ...)
-      }, error = function(e){warning(paste0("Fail for line number ", i))})
-      # Status bar
-      perc_temp <- i %% round(nrow(ldf) / n_print)
-      if(!is.na(perc_temp) & perc_temp == 0){
-        message(paste0(round(100 * i/nrow(ldf)), " % out of ", nrow(ldf),
-                       " distances calculated")) # print % of distances calculated
-      }
-    }
+  if(parallel){
+    threads <- min(c(parallel:::detectCores() * 10, n_ldf))
```

Comment:
Why is the number of threads set to parallel:::detectCores() * 10? Why do you want more threads than there are cores?